src/parser/jobs/rpr/modules/ImmortalSacrifice.tsx

```diff
@@ -0,0 +1,154 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import React from 'react'
+import {DataLink} from '../../../../components/ui/DbLink'
+import {Event, Events} from '../../../../event'
+import {Analyser} from '../../../core/Analyser'
+import {filter} from '../../../core/filter'
+import {dependency} from '../../../core/Injectable'
+import BrokenLog from '../../../core/modules/BrokenLog'
+import Checklist, {Requirement, Rule} from '../../../core/modules/Checklist'
+import {Data} from '../../../core/modules/Data'
+import {DISPLAY_ORDER} from './DISPLAY_ORDER'
+
+interface ImmortalSacrificeWindow {
+	startTime: number,
+	endTime?: number,
+	/**
+	 * If this window was closed, by consuming the stacks, or it expired
+	 */
+	closed: boolean,
+	stackCount: number,
+	consumed: boolean,
+}
+
+export default class ImmortalSacrifice extends Analyser {
+
+	static override handle = 'immortalsacrifice'
+	static override title = t('rpr.immortal-sacrifice.title')`Immortal Sacrifice stacks consumed`
+	static override debug = false
+
+	@dependency private brokenLog!: BrokenLog
+	@dependency private data!: Data
+	@dependency private checklist!: Checklist
+
+	private windows: ImmortalSacrificeWindow[] = []
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		this.addEventHook(playerFilter.type('statusApply').status(this.data.statuses.IMMORTAL_SACRIFICE.id), this.onImmortalSacrificeStack)
+		this.addEventHook(playerFilter.type('statusRemove').status(this.data.statuses.IMMORTAL_SACRIFICE.id), this.onImmortalSacrificeExpiration)
```

Comment:
Consider renaming these callbacks. Status application is not necessarily stack gain, and removal is a separate thing to expiry.

Reply:
Can you elaborate on why immortal stack is not necessary a stack gain? Because the buff to generate it has a different name, also the skill buff is different, I think you get 3 different buffs + this status when you actually get a stack, but I'm totally new on this parse stuff could be missing something.

Reply:
I'm not super across the individual statuses yet (happy to look but I have RPR at 78 only lol) but the way stacking buffs work is kinda awkward. Each stack change counts as a new status apply even when it's *lower*, so normally we'd call this something like `onGain` and reconcile stack count against `event.data`. It gets a bit weird tho since your final stack will apply with 1 and then remove, so it never actually hits zero. Sequentially, it looks something like this:

```
statusApply stacks=3
statusApply stacks=2
statusApply stacks=1
statusRemove
```

Reply:
Ah got you, but since this one you consume all the stacks at once, I think it's a bit different from the others, this you just get the statusRemove, and thats it. At least I checked multiple of my own logs and they work like that, both for consuming via plentiful harvest and from letting them expire.

Here is an example from one of my own logs
![image](https://user-images.githubusercontent.com/7723330/147860134-a48aa92c-0f97-49f9-805f-4d4ac3706b26.png)

Here is the link if you want to double-check or investigate something
https://www.fflogs.com/reports/qdYN1Q4B62RHvxM9#fight=9&type=damage-done&view=events

I did rename it as your suggestion but the logic seems good.